### ISSUE TITLE
Any chance of providing a normal Windows 11 widget?  

### ISSUE THREAD

**Issue by 0rAX0** [Khiro95/Awqat-Salaat#41]
Salam,

I really love the idea of this tasklet, it's exactly what I was looking for and it solved a major problem I had when working, so thank you for working on it. 
But I can't help but feel that there should be a proper implementation that takes advantage of the widget board on Windows 11 instead of hijacking the task bar and crashing it every time I login. Plus, it will look a lot cleaner than the way it is now

![image](https://github.com/user-attachments/assets/485910ff-1512-4847-a1dd-9ebf7263278d)

**Comment by Khiro95** [comment]
Salam @0rAX0,

I'm happy that you liked the widget and that it helped you.

The main feature of the widget is to make the prayer time, and the countdown, **always visible** to eliminate the need for extra actions (e.g. click on something) to check the time when the user need that.

Unfortunately, the widget board on Windows 11 defeats that feature. I was hoping that Microsoft allow us to use the widget button (which shows weather info) to add custom content like we have in Windows 10 and earlier (deskband feature).

I'm really sorry for the crashes and I'm trying hard to find them and fix them whenever possible. But for that I need help from the people who experience these crashes because it works fine on my machine :(
I especially need to know the steps to reproduce the crash and/or if possible some informations about what went wrong (from Event Viewer or even a crash dump).

> Plus, it will look a lot cleaner than the way it is now

Please let me know of any critic on the current way and don't hesitate to share any idea/suggestion you have to improve the widget :)

**Comment by 0rAX0** [comment]
Thanks for the detailed reply @Khiro95,

I was under the impression that you can send specific updates to the taskbar area where the widget is displaying the weather info, it's already showing updates to weather conditions, air quality, stocks, breaking news, scores...etc. What if you can send updates there for prayer times, get ready for prayer, next prayer after... etc.?

Showing the info like you are doing it right now serves my needs beautifully, so this isn't a complain, but I feel like there is a better way to do it. I even explored the possibility of trying to build the widget myself despite having 0 experience with Windows development. I guess we'll see.

In any case, I'll look into sending you crash dumps tonight inshallah

**Comment by Sheikhm007** [comment]
Assalamualaikum Koiro, could you pls bring this app to Linux (zorin os)

**Comment by Khiro95** [comment]
@0rAX0 Microsoft don't want us to touch the taskbar anymore, that's why they didn't provide a an equivalent feature to deskbands :(

And no, there is no way to update the weather widget to show custom content. So we are left with *hacks* only. That's what other widgets, such as Traffic Monitor, do by the way.

> In any case, I'll look into sending you crash dumps tonight inshallah

Much appreciated :)

**Comment by Khiro95** [comment]
Wa alaykum assalam @Sheikhm007 

I won't make any promise since I don't use Linux at all. But it's still a possibility.

Thanks for the suggestion :)

> [!note]
> **For future readers**: I won't port the widget to macOS. However, if someday I ported the widget to Linux, I will try my best to keep the code cross-platform to help those who might try to port it to macOS.

**Comment by 0rAX0** [comment]
> deskbands

Hmm, I think that I didn't make myself clear. I'm not suggesting touching the task bar, I'm suggesting plugging into what Microsoft provides already. The widget area and the widget button are very dynamic and can change to show notifications and alerts from the widgets themselves. My question was about the possibility to plug into that. Here are some examples:

![Screenshot 2024-12-18 095310](https://github.com/user-attachments/assets/231e03bc-0e32-4071-bec5-2eda38ced720)
![Screenshot 2024-12-18 095319](https://github.com/user-attachments/assets/8b3fe953-6350-4347-9230-7c3051821ed5)
![Screenshot 2024-12-18 095344](https://github.com/user-attachments/assets/486c4d67-9e9d-4c61-b8c4-4744e586ecbe)
![Screenshot 2024-12-18 095521](https://github.com/user-attachments/assets/b3ffc01a-2846-483f-b28a-2b83b1dd347b)
![Screenshot 2024-12-18 095708](https://github.com/user-attachments/assets/ea82ef2a-e7ef-4f18-a202-0f796da75d50)

As you can see, once you add cards in there, it will start showing you alerts and updates about various stuff. I usually disable everything so that I only see the weather (which also sends various updates) but once you enable everything, it becomes very dynamic. It even supports notifications inside the panel by the looks of it.

**Comment by Khiro95** [comment]
Hi @0rAX0 

Do you know if the widgets button on the taskbar can show custom content from widgets not developed by Microsoft?

If so, then in this the case the content won't be displayed *always*, which may lower the benefits of the widget. Do you know if we have control over the display time?

Sorry if these questions may look dump but I'm not a Win11 user (still using Win 10) and I'm currently busy, till the end of this month insha'Allah, so I don't have time for doing researches.

**Comment by 0rAX0** [comment]
@Khiro95 Hmm, that's a good question, I'm not sure tbh...
All I could find so far is this page: https://learn.microsoft.com/en-us/windows/apps/develop/widgets/widget-providers

I get you, I think that having in it there in the task bar is why it works well for most of us. But I'm questioning the validity of this assumption in certain cases. For example, Isha prayer time has passed, and now I'm seeing the Fajr time for tomorrow, which isn't really that helpful. Another example is seeing the Duhr time at 7 in the morning. So as you can see, the usefulness of having it there only works (for me) when it's almost time to pray, which is what the widgets allow you to do.

No worries mate, I assumed you were on Win11 already. I think I'll just dive in and try to make it myself. If I manage to do it, I'll put a link or a video for you. Otherwise, your solution is working very well. :)

**Comment by Khiro95** [comment]
Salam @0rAX0 

If I'm understanding correctly, you want the widget to be visible only when there is a certain *reasonable* time-left for the next prayer, isn't it? This could be an idea for a new feature :)

> No worries mate, I assumed you were on Win11 already. I think I'll just dive in and try to make it myself. If I manage to do it, I'll put a link or a video for you. Otherwise, your solution is working very well. :)

May Allah help you with this, and I will have a look too when I get some free time. I will keep you updated if I find anything interesting.

**Comment by 0rAX0** [comment]
> If I'm understanding correctly, you want the widget to be visible only when there is a certain _reasonable_ time-left for the next prayer, isn't it? This could be an idea for a new feature :)

Not really, if you hide while it's being the only widget in the taskbar it will leave an ugly empty space. I think that if you want to have that behavior then the official widgets are the way to go (for Win11 anyway) otherwise, the current behavior is fine. I just wish that I can find a Dump file to send you to step the crashes 😅



> May Allah help you with this, and I will have a look too when I get some free time. I will keep you updated if I find anything interesting.

Ameen. Thanks a lot!

**Comment by me-zo** [comment]
Hi there, I'm also wondering if this feature is possible. I'm ok with just hovering over the widget to see the prayer widget next to the weather widget in the widget panel.

There are tens of app on the store that do the exact same thing this app does but there are zero widgets in the widget list so i was hoping it would be possible to add it there and either have the option to toggle the app from there instead of showing it next to the weather or create the same app again but as a widget.

Either way, thank you for this app, it looks so seamless and clean with the rest of the default win stuff, I'm trying to detox from my phone this Ramadan and it helps as we don't have adhan here so i need something to track prayer time.

**Comment by Khiro95** [comment]
Hi @me-zo 

Yes, I'm still considering this feature, just lacking enough free time to work on big features.

> There are tens of app on the store that do the exact same thing this app does

You mean there are other prayer times apps, on Microsoft Store, that add a widget to the taskbar? I thought Awqat Salaat was the first.

> Either way, thank you for this app

You are welcome :)

**Comment by abdulrahman-103** [comment]
@Khiro95 it is the only one the others are windowed apps
